coral/src/app/features/approvals/schemas/SchemaApprovals.tsx

```diff
@@ -42,18 +53,112 @@ function SchemaApprovals() {
+          );
+        } else {
+          setErrorQuickActions("");
+          // If declined request is last in the page, go back to previous page
+          // This avoids staying on a non-existent page of entries, which makes the table bug hard
+          // With pagination being 0 of 0, and clicking Previous button sets active page at -1
+          // We also do not need to invalidate the query, as the activePage does not exist any more
+          // And there is no need to update anything on it
+          if (
+            schemaRequests?.entries.length === 1 &&
+            schemaRequests?.currentPage > 1
+          ) {
+            return setCurrentPage(schemaRequests?.currentPage - 1);
+          }
+
+          // We need to refetch all aclrequests queries to keep Table state in sync
+          queryClient.refetchQueries(["schemaRequestsForApprover"]);
+        }
+      },
+      onError(error: Error) {
+        setErrorQuickActions(parseErrorMsg(error));
+      },
+      onSettled() {
+        closeModal();
+      },
+    });
+
+  const { mutate: approveRequest, isLoading: approveRequestIsLoading } =
+    useMutation(approveSchemaRequest, {
+      onSuccess: (responses) => {
+        // @TODO follow up ticket #707
+        // (for all approval tables)
+        const response = responses[0];
+        if (response.result !== "success") {
+          setErrorQuickActions(
+            response.message || response.result || "Unexpected error"
+          );
+        } else {
+          setErrorQuickActions("");
+          // If declined request is last in the page, go back to previous page
+          // This avoids staying on a non-existent page of entries, which makes the table bug hard
+          // With pagination being 0 of 0, and clicking Previous button sets active page at -1
+          // We also do not need to invalidate the query, as the activePage does not exist any more
+          // And there is no need to update anything on it
+          if (
+            schemaRequests?.entries.length === 1 &&
+            schemaRequests?.currentPage > 1
+          ) {
+            return setCurrentPage(schemaRequests?.currentPage - 1);
+          }
+
+          // We need to refetch all aclrequests queries to keep Table state in sync
+          queryClient.refetchQueries(["schemaRequestsForApprover"]);
+        }
+      },
+      onError(error: Error) {
+        setErrorQuickActions(parseErrorMsg(error));
+      },
+      onSettled() {
+        closeModal();
+      },
+    });
+
   const setCurrentPage = (page: number) => {
     searchParams.set("page", page.toString());
     setSearchParams(searchParams);
   };
 
+  function closeModal() {
+    setModals({ open: "NONE", req_no: null });
+  }
+
   const table = (
     <SchemaApprovalsTable
       requests={schemaRequests?.entries || []}
-      setDetailsModal={setDetailsModal}
+      setModals={setModals}
+      quickActionLoading={approveRequestIsLoading || declineRequestIsLoading}
```

Comment:
We probably need a followup issue to implement this is other approval tables as well, right?

Reply:
Yes! Created a ticket #722

Reply:
Yes, I added a ticket for that ▶️  #710